tensorrt_llm/serve/openai_disagg_server.py

```
@@ -251,18 +251,35 @@ async def merge_streaming_responses(self, ctx_response,
+                    # Note that the retry here is needed as well as the retry in send_request function.
+                    # This is because we are using the `body_iterator` to stream the response, and it may raise an exception if the connection is lost.
+                    # The retry is needed so we send the request again to the same server.
+                    try:
+                        # Then yield the generation responses
+                        await self._increment_metric("gen_total_requests")
+                        if isinstance(gen_req, CompletionRequest):
+                            gen_response = await self.send_completion_request(gen_server, gen_req)
+                        elif isinstance(gen_req, ChatCompletionRequest):
+                            gen_response = await self.send_chat_request(gen_server, gen_req)
+                        else:
+                            raise TypeError(f"Invalid request type: {type(gen_req).__name__}")
```

Ruff (0.14.5)

Abstract `raise` to an inner function

(TRY301)